src/date-navigator/date-navigator.tsx

```diff
@@ -0,0 +1,107 @@
+import dayjs from "dayjs";
+import {
+    ArrowLeft,
+    ArrowRight,
+} from "../shared/internal-calendar/calendar-manager.style";
+import { CalendarHelper, DateHelper } from "../util";
+import {
+    Container,
+    HeaderArrowButton,
+    StyledDateText,
+    StyledDayText,
+    Wrapper,
+} from "./date-navigator.style";
+import { DateNavigatorProps } from "./types";
+
+export const DateNavigator = ({
+    selectedDate,
+    minDate,
+    maxDate,
+    isLoading,
+    onLeftArrowClick,
+    onRightArrowClick,
+    ...otherProps
+}: DateNavigatorProps) => {
+    // =============================================================================
+    // CONST, STATE, REF
+    // =============================================================================
+    const date = DateHelper.toDayjs(selectedDate);
+    const dateText = DateHelper.toDayjs(selectedDate)
+        .format("D MMMM YYYY")
+        .toString();
+    const isToday = DateHelper.isSame(selectedDate, dayjs());
+    const dayText = isToday
+        ? "Today"
+        : DateHelper.toDayjs(selectedDate).format("dddd");
+
+    // =============================================================================
+    // HELPER FUNCTIONS
+    // =============================================================================
+    const isLeftArrowDisabled = () => {
+        if (!minDate) {
+            return false;
+        }
+        return (
+            CalendarHelper.isDisabledDay(date, undefined, minDate) ||
+            DateHelper.isSame(date, minDate)
+        );
+    };
+
+    const isRightArrowDisabled = () => {
+        if (!maxDate) {
+            return false;
+        }
+        return (
+            CalendarHelper.isDisabledDay(date, undefined, undefined, maxDate) ||
+            DateHelper.isSame(date, maxDate)
+        );
+    };
+
+    // =============================================================================
+    // RENDER FUNCTIONS
+    // =============================================================================
+    return (
+        <Container {...otherProps}>
+            {onLeftArrowClick && (
+                <HeaderArrowButton
+                    id="date-navigator-left-arrow-btn-id"
+                    data-testid="date-navigator-left-arrow-btn"
+                    disabled={isLoading || isLeftArrowDisabled()}
+                    focusHighlight={false}
```

Comment:
do set `focusOutline="browser"` so that it's obvious for keyboard users when the button is focused